api/client/joinservice.go

```diff
@@ -202,6 +207,45 @@ func (c *JoinServiceClient) RegisterUsingTPMMethod(
 	return certs, nil
 }
 
+// RegisterUsingOracleMethod registers the caller using the Oracle join method and
+// returns signed certs to join the cluster. The caller must provide a
+// ChallengeResponseFunc which returns a *proto.RegisterUsingOracleMethodRequest
+// for a given challenge, or an error.
+func (c *JoinServiceClient) RegisterUsingOracleMethod(ctx context.Context, challengeResponse RegisterOracleChallengeResponseFunc) (*proto.Certs, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	oracleJoinClient, err := c.grpcClient.RegisterUsingOracleMethod(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	challengeResp, err := oracleJoinClient.Recv()
```

Comment:
challengeResp is a confusing name for this, which is supposed to hold the challenge, and when there's already a parameter named `challengeResponse`

Reply:
I'm not sure what a better name for it would be, but I renamed `challengeResponse` to hopefully reduce the confusion.